cmd/master-controller-manager/controllers.go

```diff
@@ -65,10 +67,18 @@ func createAllControllers(ctrlCtx *controllerContext) error {
 		ctrlCtx.workerNamePredicate,
 	)
 
+	configGetter, err := kubernetesprovider.DynamicKubermaticConfigurationGetterFactory(ctrlCtx.mgr.GetAPIReader(), ctrlCtx.namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get KubermaticConfiguration Getter : %w", err)
+	}
+
+	config, err := configGetter(ctrlCtx.ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get the kubermatic configuration: %w", err)
+	}
```

Comment:
If this is just a 1-time config fetch, you can use the other function in kubernetesprovider and skip the anonymous function stuff alltogether here.